chromium_src/services/network/network_context.cc

```diff
@@ -0,0 +1,33 @@
+/* Copyright (c) 2019 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "services/network/network_context.h"
+
+GURL GetURLForCookieAccess(const net::URLRequest& request) {
+  if (!request.site_for_cookies().is_empty())
+    return request.site_for_cookies();
+
+  if (request.network_isolation_key().IsFullyPopulated())
```

Comment:
we want only the top frame oringin, don't we? This way we get top frame + iframe, not sure if this is needed (https://cs.chromium.org/chromium/src/net/base/network_isolation_key.cc?sq=package:chromium&dr=CSs&g=0&l=47)

Reply:
```  // The origin of the top frame of the page making the request (where
  // applicable). Note that this is experimental and may not always be set.
  // DEPRECATED: This was introduced for the cache key and will be removed once
  // |network_isolation_key| is set for most cases.```

Reply:
I'm talking about the top frame origin contained in `network_isolation_key`. Currently `network_isolation_key().ToString()` will return slightly more than just origin.

Reply:
To be clear, I'm suggesting to use `request.network_isolation_key().GetTopFrameOrigin()`

Reply:
I see, I'll make that change